amami/__init__.py

```diff
@@ -37,11 +46,37 @@
 
 
 List of supported commands:
-------------------------------------------------------------------------------------------------
-| command | description                                                                        |
-------------------------------------------------------------------------------------------------
-| {_C_CMD}um2nc{_C_END}   | {_C_DESC}Convert a UM fieldsfile to netCDF.{_C_END}                                                 |
-| {_C_CMD}modify{_C_END}  | {_C_DESC}Modify a UM fieldsfile using data from a netCDF file, or a user-defined function.{_C_END}  |
-------------------------------------------------------------------------------------------------
+---------------------------------------------------------------------------------------------------------------------
+| {_C_BOLD}command{_C_END} | {_C_BOLD}description{_C_END}                                                                                             | 
+---------------------------------------------------------------------------------------------------------------------
+| {_C_CMD}um2nc{_C_END}   | {_C_DESC}Convert a UM fieldsfile (https://code.metoffice.gov.uk/doc/um/latest/papers/umdp_F03.pdf) to netCDF.{_C_END}    |
+---------------------------------------------------------------------------------------------------------------------
 For more information about a specific command, run `amami <command> -h`.
 """
+
+class Amami:
+    """A class that represents the `amami` application."""
+    def __init__(
+        self,
+        argv: list[str],
+    ) -> None:
+        self.args = MainParser().parse_and_process(
+            argv[1:] if argv[1:] else ["-h"]
+        )
+
+    def run_command(self):
+        """
+        Calls the entry point for the chosen`amami` command, which is the `main` 
+        function of the amami.commands.<chosen command> module.
+        """
+        command = getattr(self.args, 'subcommand')
+        command_entry_point = getattr(
+            import_module(f'amami.commands.{command}'),
+            'main',
+        )
+        # Call 'main' function of chosen command
+        command_entry_point(self.args)
+
+def main() -> None:
+    """Entry point for CLI usage of `amami`."""
+    Amami(sys.argv).run_command()
```

Comment:
The `Amami` class & `main()`  can be refactored into a single function, since the class doesn't add any functionality beyond storing args & providing a function. I'd isolate the change it to a new issue though.

Reply:
Yeah it makes sense. 

I will create an issue for this.

Reply:
Created #35